packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1571,6 +1574,91 @@ abstract class InheritedWidget extends ProxyWidget {
   bool updateShouldNotify(covariant InheritedWidget oldWidget);
 }
 
+abstract class InheritedModel<T> extends InheritedWidget {
+  const InheritedModel({ Key key, Widget child }) : super(key: key, child: child);
```

Comment:
I think InheritedModel should store a reference to a T that represents the model, and take an argument that is the value of that reference.

That is, InheritedModel<Book> would have 

```
  const InheritedModel(this.model, { Key key, Widget child});
  Book model;
```

Reply:
make sure to mention what `T` is in the docs when you write them (it's common for us to forget to document the generic types)

Reply:
T is the type of objects that represent dependencies.  I've covered T in the class doc I just added.